src/components/tasks/card/index.tsx

```diff
@@ -258,13 +262,171 @@ const Card: FC<Props> = ({
         actions.onEditRoute();
     };
 
-    return (
-        <div
-            className={`
+    // show redesign only on dev
+    if (dev)
+        return (
+            <div
+                className={`
         ${classNames.card}
+        ${classNames.card_updated}
         ${isLoading && classNames.pointerEventsNone}
         ${isTaskOverdue() && classNames.overdueTask}
     `}
+                data-testid="task-card"
+            >
+                {/* loading spinner */}
+                {isLoading && (
+                    <div className={classNames.loadingBg}>
+                        <div className={classNames.spinner}>
+                            <span className={classNames.screenReaderOnly}>
+                                loading
+                            </span>
+                        </div>
+                    </div>
+                )}
+
+                <div className={classNames.cardItems}>
+                    <h2
+                        className={classNames.cardTitle}
+                        contentEditable={shouldEdit}
+                        onKeyPress={(e) => handleChange(e, 'title')}
+                        role="button"
+                        tabIndex={0}
+                    >
+                        {cardDetails.title}
+                    </h2>
+                    {/* progress bar */}
+                    <div className={classNames.progressContainerUpdated}>
+                        <div className={classNames.progressIndicator}>
+                            <div
+                                className={`
+                ${handleProgressColor(
+                    content.percentCompleted,
+                    content.startedOn,
+                    content.endsOn
+                )}
+                ${classNames.progressStyle}
+              `}
+                                style={{
+                                    width: `${content.percentCompleted}%`,
+                                }}
+                            ></div>
+                        </div>
+                        <span>{content.percentCompleted}% </span>
+                    </div>
+                </div>
+                <div className={classNames.dateInfo}>
+                    <div>
+                        <span className={classNames.cardSpecialFont}>
+                            Estimated completion{' '}
+                        </span>
+                        {renderDate(fromNowEndsOn, shouldEdit)}
+                    </div>
+                    <span
```

Comment:
See if you can break down this JSX into multiple meaningful presentational components. Right now it is one big chunk of JSX which is hard to read and debug.

Reply:
done partially, will be tracking this with Issue #497